oref0-setup.sh fails on npm global-install

This was from a week ago or so. This was also a fresh flash of jubilinux 3.0. Other rig with existing setup was able to install just fine (also jubilinux 3.0). I've tried some of the fixes that folks mention, such as removing package-lock.json, trying the install repeating, adjusting npm proxy, nothing seems to help. @PieterGit may have insight around whether the changes to package.json may be the cause. This is from the `dev` branch
```
Checking oref0 installation
Forcing install of latest oref0 from /home/root/src/oref0/

> oref0@0.7.0-dev global-install /root/src/oref0
> npm install && sudo npm install -g && sudo npm link && sudo npm link oref0

npm WARN deprecated crypto@0.0.3: This package is no longer supported. It's now a built-in Node module. If you've depended on crypto, you should switch to the one that's built-in.
npm notice created a lockfile as package-lock.json. You should commit this file.
added 222 packages from 653 contributors and audited 362 packages in 64.01s
found 0 vulnerabilities

npm ERR! path /usr/lib/node_modules/.staging/oref0-aa44602e/node_modules/abbrev
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/usr/lib/node_modules/.staging/oref0-aa44602e/node_modules/abbrev' -> '/usr/lib/node_modules/.staging/abbrev-fd397e54'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/root/.npm/_logs/2018-11-17T06_06_45_062Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 254
npm ERR! oref0@0.7.0-dev global-install: `npm install && sudo npm install -g && sudo npm link && sudo npm link oref0`
npm ERR! Exit status 254
npm ERR!
npm ERR! Failed at the oref0@0.7.0-dev global-install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/root/.npm/_logs/2018-11-17T06_06_45_237Z-debug.log
No module named mmeowlink.vendors.mmeowlink
Installing latest mmeowlink
```

--- a/bin/oref0-log-shortcuts.sh
+++ b/bin/oref0-log-shortcuts.sh
@@ -100,7 +100,7 @@ function remove_obsolete_aliases () {
 END
 )
     echo "$OBSOLETE_ALIASES" |(while read OBSOLETE_ALIAS; do
-        cat "$PROFILE_PATH" |grep -v "$OBSOLETE_ALIAS" >"$PROFILE_PATH".new$$
+        test -f "$PROFILE_PATH" && cat "$PROFILE_PATH" |grep -v "$OBSOLETE_ALIAS" >"$PROFILE_PATH".new$$ &&
         mv -f "$PROFILE_PATH".new$$ "$PROFILE_PATH"
     done)
 }
@@ -125,10 +125,10 @@ else
     for i in "$@"; do
     case "$i" in
         --add-to-profile)
-            add_aliases_to_profile "$HOME/.bash_profile"
+            test -f "$HOME/.bash_profile" && add_aliases_to_profile "$HOME/.bash_profile"
             ;;
         --add-to-profile=*)
-            add_aliases_to_profile "${i#*=}"
+            test -f "${i#*=}" && add_aliases_to_profile "${i#*=}"
             ;;
         *)
            echo "Unrecognized argument: $i"

--- a/bin/oref0-online.sh
+++ b/bin/oref0-online.sh
@@ -49,8 +49,8 @@ main() {
                 ifdown wlan0; ifup wlan0
             fi
         fi
-        if ! ls preferences.json 2>/dev/null >/dev/null \
-            || ! cat preferences.json | jq -e .bt_with_wifi >/dev/null; then
+        if ! test -f preferences.json \
+            || ! jq -e .bt_with_wifi < preferences.json > /dev/null; then
             # if online via wifi, disconnect BT
             if has_ip wlan0 && ifconfig | egrep -q "bnep0" >/dev/null; then
                 bt_disconnect $MACs
@@ -78,8 +78,8 @@ main() {
         if ! has_ip wlan0; then
             wifi_dhcp_renew
         fi
-        if ! ls preferences.json 2>/dev/null >/dev/null \
-            || ! cat preferences.json | jq -e .bt_with_wifi >/dev/null; then
+        if ! test -f preferences.json \
+            || ! jq -e .bt_with_wifi < preferences.json > /dev/null; then
             if ! check_ip >/dev/null; then
                 bt_connect $MACs
             fi
@@ -149,25 +149,30 @@ function bt_connect {
     for MAC; do
         #echo -n "At $(date) my public IP is: "
         if ! check_ip >/dev/null \
-               || (ls preferences.json 2>/dev/null >/dev/null \
-                && cat preferences.json | jq -e .bt_with_wifi >/dev/null); then
+               || (test -f preferences.json \
+                && jq -e .bt_with_wifi < preferences.json > /dev/null); then
             echo; echo "No Internet access detected, attempting to connect BT to $MAC"
             oref0-bluetoothup
-            if ! ls preferences.json 2>/dev/null >/dev/null \
-                || ! cat preferences.json | jq -e .bt_offline >/dev/null \
+            if ! test -f preferences.json \
+                || ! jq -e .bt_offline < preferences.json > /dev/null \
                 || ! ifconfig | egrep -q "bnep0" >/dev/null; then
                 echo "Attempting to connect to bt $MAC..."
                 sudo bt-pan client $MAC -d
-            for i in {1..3}
-            do
+              for i in {1..3}
+              do
                 sudo bt-pan client $MAC
                 sudo dhclient bnep0
-            done
-                echo "...done."
+              done
+                if ! has_ip bnep0; then
+                  bt_bnep0_cycle
+                fi
             else
                 echo "Preference bt_offline enabled - already have a bt interface, not reconnecting it."
                 sudo dhclient bnep0
                 ifconfig | egrep -A6 "wlan|bnep"
+                if ! has_ip bnep0; then
+                  bt_bnep0_cycle
+                fi
             fi
             if ifconfig | egrep -q "bnep0" >/dev/null; then
                 echo -n "Connected to Bluetooth with IP: "
@@ -175,8 +180,8 @@ function bt_connect {
             fi
             # if we couldn't reach the Internet over wifi, but (now) have a bnep0 IP, release the wifi IP/route
             if has_ip wlan0 && has_ip bnep0 && ! grep -q $HostAPDIP /etc/network/interfaces; then
-                   if ! ls preferences.json 2>/dev/null >/dev/null \
-                   || ! cat preferences.json | jq -e .bt_with_wifi >/dev/null; then
+                   if ! test -f preferences.json \
+                   || ! jq -e .bt_with_wifi < preferences.json > /dev/null; then
                     echo "Have bt connection, releasing wifi"
                     # release the wifi IP/route but *don't* renew it, in case it's not working
                     sudo dhclient wlan0 -r
@@ -236,6 +241,12 @@ function stop_cycle {
     stop_hotspot
     echo "Cycling wlan0"
     ifdown wlan0; ifup wlan0
+}
+
+function bt_bnep0_cycle {
+  echo -n "No IP address assigned, cycling the bnep0 interface"
+  sudo ifdown bnep0; sudo ifup bnep0;
+  echo "...done" 
 }
 
 

--- a/bin/oref0-setup.sh
+++ b/bin/oref0-setup.sh
@@ -720,7 +720,7 @@ if prompt_yn "" N; then
         )
     fi
 
-    cp preferences.json old_preferences.json
+    test -f preferences.json && cp preferences.json old_preferences.json || echo "No old preferences.json to save off"
     if [[ "$max_iob" == "0" && -z "$max_daily_safety_multiplier" && -z "$current_basal_safety_multiplier" && -z "$min_5m_carbimpact" ]]; then
         oref0-get-profile --exportDefaults > preferences.json || die "Could not run oref0-get-profile"
     else
@@ -780,8 +780,9 @@ if prompt_yn "" N; then
         set_pref_string .cgm_loop_path "$directory-cgm-loop"
     fi
 
-    if [[ ${CGM,,} =~ "xdrip" || ${CGM,,} =~ "xdrip-js" ]]; then
+    if [[ ${CGM,,} =~ "xdrip" ]]; then # Evaluates true for both xdrip and xdrip-js 
         set_pref_string .xdrip_path "$HOME/.xDripAPS"
+        set_pref_string .bt_offline "true"
     fi
     #if [[ ! -z "$DEXCOM_CGM_TX_ID" ]]; then
         #set_pref_string .dexcom_cgm_tx_id "$DEXCOM_CGM_TX_ID"
@@ -851,6 +852,11 @@ if prompt_yn "" N; then
 
     echo Checking for BT Mac, BT Peb, Shareble, or xdrip-js
     if [[ ! -z "$BT_PEB" || ! -z "$BT_MAC" || ! -z $BLE_SERIAL || ! -z $DEXCOM_CGM_TX_ID ]]; then
+        if [ ! -z "$BT_MAC" ]; then
+          printf 'Checking for the bnep0 interface in the interfaces files and adding if missing...'
+          # Make sure the bnep0 interface is in the /etc/networking/interface
+          (grep -qa bnep0 /etc/network/interfaces && printf 'skipped.\n') || (printf '\n%s\n\n' "iface bnep0 inet dhcp" >> /etc/network/interfaces && printf 'added.\n') 
+        fi
         # Install Bluez for BT Tethering
         echo Checking bluez installation
         bluetoothdversion=$(bluetoothd --version || 0)

--- a/bin/oref0-setup.sh
+++ b/bin/oref0-setup.sh
@@ -853,7 +853,7 @@ if prompt_yn "" N; then
     echo Checking for BT Mac, BT Peb, Shareble, or xdrip-js
     if [[ ! -z "$BT_PEB" || ! -z "$BT_MAC" || ! -z $BLE_SERIAL || ! -z $DEXCOM_CGM_TX_ID ]]; then
         if [ ! -z "$BT_MAC" ]; then
-          printf 'Checking for the bnep0 interface in the interfaces files and adding if missing...'
+          printf 'Checking for the bnep0 interface in the interfaces file and adding if missing...'
           # Make sure the bnep0 interface is in the /etc/networking/interface
           (grep -qa bnep0 /etc/network/interfaces && printf 'skipped.\n') || (printf '\n%s\n\n' "iface bnep0 inet dhcp" >> /etc/network/interfaces && printf 'added.\n') 
         fi

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "openaps oref0 reference implementation of the reference design",
   "scripts": {
     "test": "make test",
-    "global-install": "npm install && sudo npm install -g && sudo npm link && sudo npm link oref0"
+    "global-install": "npm install && sudo npm link && sudo npm link oref0 && sudo npm install -g"
   },
   "repository": {
     "type": "git",

